certbot/main.py

```diff
@@ -678,7 +678,8 @@ def main(cli_args=sys.argv[1:]):
     elif config.text_mode:
         displayer = display_util.FileDisplay(sys.stdout)
     elif config.verb == "renew":
-        config.noninteractive_mode = True
+        if not config.gui:
```

Comment:
This shouldn't just apply to `renew`. While redundant, people should be able `--dialog` for other subcommands as well.